backend/dal/Repositories/DocumentActivityRepository.cs

```diff
@@ -34,32 +34,38 @@ public DocumentActivityRepository(PimsContext dbContext, ClaimsPrincipal user, I
         /// Get a list of all the document relationships for a given document.
         /// </summary>
         /// <returns></returns>
-        public IList<PimsActivityInstanceDocument> GetAllByDocument(long documentId)
+        public IList<PimsActivityInstanceDocument> GetAllByDocument(long documentId, bool tracking = false)
         {
-            return this.Context.PimsActivityInstanceDocuments
+            var query = this.Context.PimsActivityInstanceDocuments
                 .Include(ad => ad.Document)
                     .ThenInclude(d => d.DocumentStatusTypeCodeNavigation)
                 .Include(ad => ad.Document)
                     .ThenInclude(d => d.DocumentType)
-                .Where(ad => ad.DocumentId == documentId)
-                .AsNoTracking()
-                .ToList();
+                .Where(ad => ad.DocumentId == documentId);
+            if (tracking)
+            {
+                query = query.AsNoTracking();
+            }
```

Comment:
we should standardize this so the repositories are agnostic about the usage

Reply:
Also the variable name is confusing - so if tracking == true we want the query as NO TRACKING? Or is it a bug?

Reply:
I removed this, need to discuss our current deletion pattern on Friday, our current pattern does not work for multiple deletes in a single transaction.